### PR TITLE
feat: Missing manuscripts.

### PR DESCRIPTION
**Pull request by Yedson54**
The missing manuscript option allows deleting some manuscripts from the stemma. It requires specifying the proportion of manuscripts to delete in the configuration parser.

* src/stemmabench/config_parser.py
    - Add `missing_manuscripts` field to StemmaConfig.

* src/stemmabench/stemma_generator.py
    - Add a `missing_manuscripts` method in `Stemma` class.
    - Create a missing tradition in output_folder when using CLI.

* demo/config.yaml
* src/tests/test_data/config.yaml
    - Update configuration files to account for missing manuscripts.

* src/stemmabench/textual_units/word.py
    - fixing `mispell`

### CODE REVIEW

File: src/stemmabench/stemma_generator.py [hunk 1]
```diff
@@ -34,9 +36,11 @@ def __init__(
         else:
             self.config = StemmaBenchConfig.from_yaml(config_path)
         self.depth = self.config.stemma.depth
+        self.missing_manuscripts_rate = self.config.stemma.missing_manuscripts.rate
         self._levels: List[Dict[str, List[str]]] = []
         self.texts_lookup = {}
         self.edges = []
+        self.random_state = random_state
```
Comment (SphRbtHyk) [inline src/stemmabench/stemma_generator.py:43]: Is this for testing purpose ? If so, can you add an inline comment ?

File: src/stemmabench/stemma_generator.py [hunk 2]
```diff
@@ -19,6 +20,7 @@ def __init__(
         config_path: str = None,
         original_text: str = None,
         path_to_text: str = None,
+        random_state: int = None
     ) -> None:
         """A class to perform variant generation.
```
Comment (SphRbtHyk) [inline src/stemmabench/stemma_generator.py:25]: Absolutely not your fault, but can you add the docstrings here please ?

File: src/stemmabench/stemma_generator.py [hunk 3]
```diff
@@ -90,9 +94,29 @@ def __repr__(self) -> str:
...
         return [Text(manuscript).transform(self.config.variants,
-                                           meta_config=self.config.meta) 
+                                           meta_config=self.config.meta)
                                            for _ in range(self.width)]
 
+    def missing_manuscripts(self) -> Tuple[Dict[str, str], List[Tuple[str]]]:
+        """Remove some manuscripts from the tradition.
+        """
+        # Initialize a RNG.
+        rng = np.random.default_rng(self.random_state)
```
Comment (SphRbtHyk) [inline src/stemmabench/stemma_generator.py:104]: why is this required ? Won't this affect the randomness of the process ?

**Comment by SphRbtHyk** [comment]
Can you please add a section in the documentation about the missing manuscript process ?

File: src/stemmabench/stemma_generator.py (hunk 3 above)
Comment (Yedson54) [reply src/stemmabench/stemma_generator.py:104]: No, as long as it's set to "None," the "random_state" does not influence the randomness of the process.

Comment (SphRbtHyk) [reply src/stemmabench/stemma_generator.py:104]: - [ ] Replace rng with choice

File: src/stemmabench/stemma_generator.py (hunk 2 above)
Comment (Yedson54) [reply src/stemmabench/stemma_generator.py:25]: - [x] Complete docstrings.

File: src/stemmabench/stemma_generator.py (hunk 1 above)
Comment (Yedson54) [reply src/stemmabench/stemma_generator.py:43]: - [x] Remove `random_state` property.

File: docs/variants.md [hunk 4]
```diff
@@ -56,7 +59,11 @@ Possible variants are:
 ### Fragmentation
 
 ### Missing manuscripts
+In the StemmaBench package, the concept of missing manuscripts is modeled using a probabilistic process governed by a p-parameter Bernoulli distribution. After having generating the whole tradition, a proportion *p* of the manuscripts are selected (with equiprobability) and deleted from the tradition. This simulates the loss or absence of certain manuscripts over time.
```
Comment (SphRbtHyk) [inline docs/variants.md:62]: I think simply mentioning "loss" instead of "loss or absence" is enough.

File: src/stemmabench/stemma_generator.py [hunk 5]
```diff
@@ -131,6 +166,7 @@ def generate(self):
             level_name += f":{self.depth - remaining_depth - 1}"
             # Decrease remaining depth
             remaining_depth -= 1
+        # Missing manuscript.
```
Comment (SphRbtHyk) [inline src/stemmabench/stemma_generator.py:169]: I am not sure what this comment refers to ?

**Review by SphRbtHyk — APPROVED** [review]

**Comment by SphRbtHyk** [comment]
One major question: where is this feature tested ?

File: docs/variants.md (hunk 4 above)
Comment (Yedson54) [reply docs/variants.md:62]: - [x] "loss or absence" replace by "loss".

File: src/stemmabench/stemma_generator.py (hunk 5 above)
Comment (Yedson54) [reply src/stemmabench/stemma_generator.py:169]: - [x] Remove unecessary comment.

I was intending to incorporate an missing manuscript step during the generation process, thereby altering the existing output stemma property (for RAM use). But finally, it didn't serve.
